rfc#00002

```diff
@@ -0,0 +1,93 @@
+# Title (Ex. 00002: Trust domain introduction)
+- Authors: [Mike Bursell](you@github-email)
+- Status: [PROPOSED](/README.md#proposed)
+- Since: 2020-02-13
+- Status Note: under discussion
+- Start Date: 2020-02-11
+- Tags: trust
+
+## Summary
+
+Trust domains are a key concept within Enarx.  This RFC provides an
+introduction to trust domains.
+
+## Motivation
+
+In order to explain the motivations behind Enarx, our design principles
+and aims, we need to be able to explain what trust domains are.  This RFC
+attempts to provide an introduction.
+
+## Tutorial
+
+A definition of trust, from [What is trust?](http://aliceevebob.com/2017/05/09/what-is-trust-with-apologies-to-pontius-pilate/).
+
+"Trust is the assurance that one entity holds that another will
+perform particular actions according to a specific expectation."
+First corollary: "Trust is always contextual."
+Second corollary: "One of the contexts for trust is always time."
+Third corollary: "Trust relationships are not symmetrical."
+
+### Applying this to Enarx
+The key trust relationship for Enarx is that held by the tenant
+(workload owner) to that workload.  Enarx aims to provide assurances to
+the tenant that the integrity and confidentiality of that workload are
+protected: these are the key contexts of that trust, and actions can be
+assured if they are maintained.  Trust may decay or be built up in the
+workload - the time context.  In this case, the workload does not have an
+equivalent set of assurances about the tenant's integrity and
+confidentiality, so the relationship if there were one, would not be
+symmetrical.
+
+### Trust domains
```

Comment:
Something i'd like to see in this section is how trust domains are, essentially (and if i've understood correctly), "what needs to be considered from a trust-perspective by actor X", or in other words what X should worry about.

Another thing i think would be useful is how trust domains are in interesting concept as they allow to think in very obvious terms about what is in them, but also (and this is important), what moves in and out of them. This last aspect is crucial to understanding how Enarx Keeps are built and what trust relationships exist (and how they change) during this build-up process.

Reply:
Yes, I want to provide an RFC on this specifically. An issue exists.